### ISSUE TITLE
Fix: Error signal averaging should use summing for gradient accumulation

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1651]
## Context

Part of #1649

In `src/propagate/TopologicalBackpropagation.ts` line 93, the error signal from multiple downstream neurons is **averaged**:

```typescript
const averageDelta = targetDeltaSum[neuronIndex] / downstreamCount;
```

In standard backpropagation, gradients from multiple downstream paths are **summed**, not averaged. When a neuron feeds 10 downstream neurons that each request an activation increase of 0.1, the total gradient should be 1.0, but averaging reduces it to 0.1. This loses gradient magnitude for highly-connected neurons, which directly impacts large production creatures (976 neurons, 18,605 synapses) where neurons have many downstream connections.

This is a likely root cause for why "backpropagation logic hasn't been finding improvements in the score of our large creatures to the same extent as the other evolutionary approaches."

The original comment says "This matches the per-visit error magnitude from the recursive approach" — but the recursive approach processed each visit independently and accumulated weight/bias adjustments across all visits. The topological approach should accumulate the **total** error, not the average.

## What needs to be done

1. Change the error distribution to use the **sum** of target deltas instead of the average
2. Remove the `downstreamCount` division on line 93
3. The `downstreamCount` repetition loops (lines 285 and 308) should also be removed as part of this change, since they were compensating for the averaging by repeating accumulation — see the related issue below
4. Adjust tests to verify gradient magnitude scales correctly with fan-out

## Acceptance criteria

- [ ] Error signal uses summed deltas, not averaged
- [ ] Tests demonstrate that a neuron feeding N downstream paths receives N× the error signal of one feeding 1 path (given equal per-path error)
- [ ] Convergence tests pass (existing tests in `test/propagate/TopologicalBackpropagation.ts`)
- [ ] A benchmark on a production-sized creature (observations ~2356, neurons ~976, synapses ~18605) shows equal or improved convergence compared to the current averaging approach

## Dependencies

- Should be coordinated with the `downstreamCount` repetition removal (they are related compensating mechanisms)

## Approach

This is a logic fix — follow TDD as specified in the parent issue.
